### PR TITLE
feat: [CDS-75108]: fetching controller using java client for pod detail.

### PR DESCRIPTION
**Pull request by GoelPratyush99**
## Describe your changes
fetching controller using java client to update the service name in pod detail

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/55115)

### CODE REVIEW

**Comment by GoelPratyush99** [comment]
trigger uts

File: 260-delegate/src/main/java/io/harness/perpetualtask/instancesyncv2/cg/CgK8sInstancesDetailsFetcher.java [hunk 1]
```diff
@@ -247,16 +251,47 @@ private List<ContainerInfo> fetchRunningK8sPodsForContainerServiceName(
...
+                                                                 .map(entry -> entry.getKey() + "=" + entry.getValue())
+                                                                 .collect(Collectors.joining(","));
+        V1ServiceList serviceList = kubernetesContainerService.getServiceList(kubernetesConfig, serviceLabelString);
+        if (isNotEmpty(serviceList.getItems()) && isNotEmpty(serviceList.getItems().get(0).getMetadata().getName())) {
+          serviceName = serviceList.getItems().get(0).getMetadata().getName();
+        } else {
+          serviceName = "None";
+        }
+      }
+    } catch (Exception exception) {
```
Comment (abosii-harness) [inline 260-delegate/src/main/java/io/harness/perpetualtask/instancesyncv2/cg/CgK8sInstancesDetailsFetcher.java:279]: Let's log that we have reached fallback so we could know and debug if something is wrong

Comment (abosii-harness) [reply 260-delegate/src/main/java/io/harness/perpetualtask/instancesyncv2/cg/CgK8sInstancesDetailsFetcher.java:279]: Actually is better to have a flag that we can use to check if the k8s java client failed but fabric8 is successful

**Review by abhinavsingh1196 — CHANGES_REQUESTED** [review]
> Reviewed 9 of 9 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55115)* status: 9 file(s) to review. 1 change requests. 0 of 2 approvals obtained. Waiting on team(s): k8s-team. 

___
*[`260-delegate/src/main/java/io/harness/perpetualtask/instancesyncv2/cg/CgK8sInstancesDetailsFetcher.java` line 65 at r1](https://reviewable.io/reviews/harness/harness-core/55115#-NispnpI66qbYTgl-Ses:-NispnpI66qbYTgl-Set:b-ltanfa) ([raw file](https://github.com/harness/harness-core/blob/69455926f9dfac4c094edbc27f95d509f36c8d84/260-delegate/src/main/java/io/harness/perpetualtask/instancesyncv2/cg/CgK8sInstancesDetailsFetcher.java#L65)):*
> ```Java
> import lombok.RequiredArgsConstructor;
> import lombok.extern.slf4j.Slf4j;
> import org.jooq.tools.StringUtils;
> ```

nit: use apache commons StringUtils for consistency
___
*[`260-delegate/src/main/java/io/harness/perpetualtask/instancesyncv2/cg/CgK8sInstancesDetailsFetcher.java` line 260 at r1](https://reviewable.io/reviews/harness/harness-core/55115#-NispYGDDP76inr6Jiif:-NispYGDDP76inr6Jiig:brhsp9d) ([raw file](https://github.com/harness/harness-core/blob/69455926f9dfac4c094edbc27f95d509f36c8d84/260-delegate/src/main/java/io/harness/perpetualtask/instancesyncv2/cg/CgK8sInstancesDetailsFetcher.java#L260)):*
> ```Java
>       V1ObjectMeta controller = kubernetesContainerService.getController(kubernetesConfig, containerServiceName);
>       if (controller != null) {
>         controllerName = controller.getName();
> ```

also would suggest to break this method into two methods like 
`getServiceNameUsingFabric8` & `getServiceNameUsingK8sClient`
___
*[`260-delegate/src/main/java/io/harness/perpetualtask/instancesyncv2/cg/CgK8sInstancesDetailsFetcher.java` line 276 at r1](https://reviewable.io/reviews/harness/harness-core/55115#-Nimfn8-4o5dpoHAF_-d:-Nimfn8-4o5dpoHAF_-e:b-axlqtr) ([raw file](https://github.com/harness/harness-core/blob/69455926f9dfac4c094edbc27f95d509f36c8d84/260-delegate/src/main/java/io/harness/perpetualtask/instancesyncv2/cg/CgK8sInstancesDetailsFetcher.java#L276)):*
> ```Java
>           serviceName = serviceList.getItems().get(0).getMetadata().getName();
>         } else {
>           serviceName = "None";
> ```

nit: can just initialize serviceName as "None" above to get rid of this else block?
___
*[`930-delegate-tasks/src/main/java/software/wings/service/impl/ContainerServiceImpl.java` line 71 at r1](https://reviewable.io/reviews/harness/harness-core/55115#-Nispu627QhodsYsEfPv:-NimgwXq5LcQ-mXSYln6:b-acwl86) ([raw file](https://github.com/harness/harness-core/blob/69455926f9dfac4c094edbc27f95d509f36c8d84/930-delegate-tasks/src/main/java/software/wings/service/impl/ContainerServiceImpl.java#L71)):*
> ```Java
> import lombok.extern.slf4j.Slf4j;
> import org.apache.commons.lang3.tuple.Pair;
> import org.jooq.tools.StringUtils;
> ```

nit: use apache commons for consistency
___
*[`930-delegate-tasks/src/main/java/software/wings/service/impl/ContainerServiceImpl.java` line 149 at r1](https://reviewable.io/reviews/harness/harness-core/55115#-NispzXr37w9uc5RIvMZ:-NispzXr37w9uc5RIvM_:bmih2cx) ([raw file](https://github.com/harness/harness-core/blob/69455926f9dfac4c094edbc27f95d509f36c8d84/930-delegate-tasks/src/main/java/software/wings/service/impl/ContainerServiceImpl.java#L149)):*
> ```Java
>           V1ObjectMeta controller = kubernetesContainerService.getController(kubernetesConfig, containerServiceName);
>           if (controller != null) {
>             controllerName = controller.getName();
> ```

Looks the same as `CgK8sInstancesDetailsFetcher#fetchRunningK8sPodsForContainerServiceName`. If duplicate, should be extracted
___
*[`960-api-services/src/main/java/io/harness/k8s/KubernetesContainerService.java` line 56 at r1](https://reviewable.io/reviews/harness/harness-core/55115#-NisqMD18_8l51_3dUS9:-NisqMD18_8l51_3dUSA:bg40dhx) ([raw file](https://github.com/harness/harness-core/blob/69455926f9dfac4c094edbc27f95d509f36c8d84/960-api-services/src/main/java/io/harness/k8s/KubernetesContainerService.java#L56)):*
> ```Java
>   HasMetadata createOrReplaceController(KubernetesConfig kubernetesConfig, HasMetadata definition);
> 
>   HasMetadata getFabric8Controller(KubernetesConfig kubernetesConfig, String name);
> ```

nit: similar methods, should name consistently with same prefix like
getControllerK8s/getControllerFabric8
___
*[`960-api-services/src/main/java/io/harness/k8s/KubernetesContainerService.java` line 109 at r1](https://reviewable.io/reviews/harness/harness-core/55115#-NisqXlj52ZVkbWU031p:-NisqXlj52ZVkbWU031q:b-retqtu) ([raw file](https://github.com/harness/harness-core/blob/69455926f9dfac4c094edbc27f95d509f36c8d84/960-api-services/src/main/java/io/harness/k8s/KubernetesContainerService.java#L109)):*
> ```Java
>   V1Service getService(KubernetesConfig kubernetesConfig, String name);
> 
>   V1ServiceList getServiceList(KubernetesConfig kubernetesConfig, String labelSelector);
> ```

nit: again, similar methods, so should be named consistently
getServicesUsingK8sClient/getServicesUsingFabric8
___
*[`960-api-services/src/main/java/io/harness/k8s/KubernetesContainerServiceImpl.java` line 411 at r1](https://reviewable.io/reviews/harness/harness-core/55115#-Nissbu44l0FXK0jJEVm:-Nissbu44l0FXK0jJEVn:bw4knfg) ([raw file](https://github.com/harness/harness-core/blob/69455926f9dfac4c094edbc27f95d509f36c8d84/960-api-services/src/main/java/io/harness/k8s/KubernetesContainerServiceImpl.java#L411)):*
> ```Java
>       V1ReplicationControllerList v1ReplicationControllerList =
>           new CoreV1Api(apiClient).listNamespacedReplicationController(
>               kubernetesConfig.getNamespace(), null, null, null, null, null, null, null, null, null, null);
> ```

should we retry these like other K8s API calls? 
___
*[`960-api-services/src/main/java/io/harness/k8s/KubernetesContainerServiceImpl.java` line 432 at r1](https://reviewable.io/reviews/harness/harness-core/55115#-Niss7T-21-M5CUrP_oX:-Niss7T-21-M5CUrP_oY:b-ck354b) ([raw file](https://github.com/harness/harness-core/blob/69455926f9dfac4c094edbc27f95d509f36c8d84/960-api-services/src/main/java/io/harness/k8s/KubernetesContainerServiceImpl.java#L432)):*
> ```Java
>       } catch (ApiException exception) {
>         if (!isResourceNotFoundException(exception.getCode())) {
>           String message = format("Unable to get %s/Deployment. Code: %s, message: %s", kubernetesConfig.getNamespace(),
> ```

nit: type precedes name usually, 
`Unable to get Deployment/%s`
___
*[`960-api-services/src/main/java/io/harness/k8s/KubernetesContainerServiceImpl.java` line 448 at r1](https://reviewable.io/reviews/harness/harness-core/55115#-NisrwuN5jtITbA1ttQm:-NisrwuN5jtITbA1ttQn:b-76pxn5) ([raw file](https://github.com/harness/harness-core/blob/69455926f9dfac4c094edbc27f95d509f36c8d84/960-api-services/src/main/java/io/harness/k8s/KubernetesContainerServiceImpl.java#L448)):*
> ```Java
>       } catch (ApiException exception) {
>         if (!isResourceNotFoundException(exception.getCode())) {
>           String message = format("Unable to get %s/Deployment. Code: %s, message: %s", kubernetesConfig.getNamespace(),
> ```

ReplicaSet
___
*[`960-api-services/src/main/java/io/harness/k8s/KubernetesContainerServiceImpl.java` line 480 at r1](https://reviewable.io/reviews/harness/harness-core/55115#-NissCj1DWJ0IxFdITfv:-NissCj1DWJ0IxFdITfw:bbh3m49) ([raw file](https://github.com/harness/harness-core/blob/69455926f9dfac4c094edbc27f95d509f36c8d84/960-api-services/src/main/java/io/harness/k8s/KubernetesContainerServiceImpl.java#L480)):*
> ```Java
>       } catch (ApiException exception) {
>         if (!isResourceNotFoundException(exception.getCode())) {
>           String message = format("Unable to get %s/Stateful Set. Code: %s, message: %s",
> ```

DaemonSet
___
*[`960-api-services/src/main/java/io/harness/k8s/KubernetesContainerServiceImpl.java` line 1420 at r1](https://reviewable.io/reviews/harness/harness-core/55115#-Nist1ZPBtojL0CW6mfL:-Nist1ZPBtojL0CW6mfM:b-6e178u) ([raw file](https://github.com/harness/harness-core/blob/69455926f9dfac4c094edbc27f95d509f36c8d84/960-api-services/src/main/java/io/harness/k8s/KubernetesContainerServiceImpl.java#L1420)):*
> ```Java
>           return null;
>         }
>         String message = format("Unable to get %s/Service with labelSelector %s. Code: %s, message: %s",
> ```

nit: should be Service/%s
___
*[`960-api-services/src/test/java/io/harness/k8s/KubernetesContainerServiceImplTest.java` line 1948 at r1](https://reviewable.io/reviews/harness/harness-core/55115#-NistNw04gKfZldu-FaM:-NistNw04gKfZldu-FaN:b-ouask) ([raw file](https://github.com/harness/harness-core/blob/69455926f9dfac4c094edbc27f95d509f36c8d84/960-api-services/src/test/java/io/harness/k8s/KubernetesContainerServiceImplTest.java#L1948)):*
> ```Java
>     V1ObjectMeta controller = kubernetesContainerService.getController(KUBERNETES_CONFIG, "controllerName");
>     assertThat(controller).isNotNull();
>   }
> ```

should we also add test for 
- when all API calls fail and it returns null
- for customObjects API
?

File: 260-delegate/src/main/java/io/harness/perpetualtask/instancesyncv2/cg/CgK8sInstancesDetailsFetcher.java (hunk 1 above)
Comment (GoelPratyush99) [reply 260-delegate/src/main/java/io/harness/perpetualtask/instancesyncv2/cg/CgK8sInstancesDetailsFetcher.java]: done

**Review by GoelPratyush99 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55115)* status: 10 file(s) to review. 1 change requests. 7. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`260-delegate/src/main/java/io/harness/perpetualtask/instancesyncv2/cg/CgK8sInstancesDetailsFetcher.java` line 65 at r1](https://reviewable.io/reviews/harness/harness-core/55115#-NispnpI66qbYTgl-Ses:-NjgcV_N0TyvuxilQ4J3:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/69455926f9dfac4c094edbc27f95d509f36c8d84/260-delegate/src/main/java/io/harness/perpetualtask/instancesyncv2/cg/CgK8sInstancesDetailsFetcher.java#L65)):*
<details><summary><i>Previously, abhinavsingh1196 (Abhinav Singh) wrote…</i></summary><blockquote>

nit: use apache commons StringUtils for consistency
</blockquote></details>

done
___
*[`260-delegate/src/main/java/io/harness/perpetualtask/instancesyncv2/cg/CgK8sInstancesDetailsFetcher.java` line 260 at r1](https://reviewable.io/reviews/harness/harness-core/55115#-NispYGDDP76inr6Jiif:-NjgcXcH6VuoWWkU0GEh:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/69455926f9dfac4c094edbc27f95d509f36c8d84/260-delegate/src/main/java/io/harness/perpetualtask/instancesyncv2/cg/CgK8sInstancesDetailsFetcher.java#L260)):*
<details><summary><i>Previously, abhinavsingh1196 (Abhinav Singh) wrote…</i></summary><blockquote>

also would suggest to break this method into two methods like 
`getServiceNameUsingFabric8` & `getServiceNameUsingK8sClient`
</blockquote></details>

done
___
*[`260-delegate/src/main/java/io/harness/perpetualtask/instancesyncv2/cg/CgK8sInstancesDetailsFetcher.java` line 276 at r1](https://reviewable.io/reviews/harness/harness-core/55115#-Nimfn8-4o5dpoHAF_-d:-NjgcYR01-XOrUwZTkCZ:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/69455926f9dfac4c094edbc27f95d509f36c8d84/260-delegate/src/main/java/io/harness/perpetualtask/instancesyncv2/cg/CgK8sInstancesDetailsFetcher.java#L276)):*
<details><summary><i>Previously, abhinavsingh1196 (Abhinav Singh) wrote…</i></summary><blockquote>

nit: can just initialize serviceName as "None" above to get rid of this else block?
</blockquote></details>

done
___
*[`930-delegate-tasks/src/main/java/software/wings/service/impl/ContainerServiceImpl.java` line 71 at r1](https://reviewable.io/reviews/harness/harness-core/55115#-Nispu627QhodsYsEfPv:-NjgcbEKBHn1XLueX1nd:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/69455926f9dfac4c094edbc27f95d509f36c8d84/930-delegate-tasks/src/main/java/software/wings/service/impl/ContainerServiceImpl.java#L71)):*
<details><summary><i>Previously, abhinavsingh1196 (Abhinav Singh) wrote…</i></summary><blockquote>

nit: use apache commons for consistency
</blockquote></details>

done
___
*[`930-delegate-tasks/src/main/java/software/wings/service/impl/ContainerServiceImpl.java` line 149 at r1](https://reviewable.io/reviews/harness/harness-core/55115#-NispzXr37w9uc5RIvMZ:-Njgck5ICpOoW81y1KGP:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/69455926f9dfac4c094edbc27f95d509f36c8d84/930-delegate-tasks/src/main/java/software/wings/service/impl/ContainerServiceImpl.java#L149)):*
<details><summary><i>Previously, abhinavsingh1196 (Abhinav Singh) wrote…</i></summary><blockquote>

Looks the same as `CgK8sInstancesDetailsFetcher#fetchRunningK8sPodsForContainerServiceName`. If duplicate, should be extracted
</blockquote></details>

done
___
*[`960-api-services/src/main/java/io/harness/k8s/KubernetesContainerService.java` line 56 at r1](https://reviewable.io/reviews/harness/harness-core/55115#-NisqMD18_8l51_3dUS9:-NjgcncI0XDavVXp_bWI:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/69455926f9dfac4c094edbc27f95d509f36c8d84/960-api-services/src/main/java/io/harness/k8s/KubernetesContainerService.java#L56)):*
<details><summary><i>Previously, abhinavsingh1196 (Abhinav Singh) wrote…</i></summary><blockquote>

nit: similar methods, should name consistently with same prefix like
getControllerK8s/getControllerFabric8
</blockquote></details>

done
___
*[`960-api-services/src/main/java/io/harness/k8s/KubernetesContainerServiceImpl.java` line 411 at r1](https://reviewable.io/reviews/harness/harness-core/55115#-Nissbu44l0FXK0jJEVm:-Njgcu6c4zRW0nwNDpQ4:b-ryx179) ([raw file](https://github.com/harness/harness-core/blob/69455926f9dfac4c094edbc27f95d509f36c8d84/960-api-services/src/main/java/io/harness/k8s/KubernetesContainerServiceImpl.java#L411)):*
<details><summary><i>Previously, abhinavsingh1196 (Abhinav Singh) wrote…</i></summary><blockquote>

should we retry these like other K8s API calls?
</blockquote></details>

We not using retry to for fabric8 case
___
*[`960-api-services/src/main/java/io/harness/k8s/KubernetesContainerServiceImpl.java` line 432 at r1](https://reviewable.io/reviews/harness/harness-core/55115#-Niss7T-21-M5CUrP_oX:-NjgdIVn8QpS3nF_Rsqg:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/69455926f9dfac4c094edbc27f95d509f36c8d84/960-api-services/src/main/java/io/harness/k8s/KubernetesContainerServiceImpl.java#L432)):*
<details><summary><i>Previously, abhinavsingh1196 (Abhinav Singh) wrote…</i></summary><blockquote>

nit: type precedes name usually, 
`Unable to get Deployment/%s`
</blockquote></details>

done
___
*[`960-api-services/src/main/java/io/harness/k8s/KubernetesContainerServiceImpl.java` line 448 at r1](https://reviewable.io/reviews/harness/harness-core/55115#-NisrwuN5jtITbA1ttQm:-NjgdJ_228bmNFdDLl6w:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/69455926f9dfac4c094edbc27f95d509f36c8d84/960-api-services/src/main/java/io/harness/k8s/KubernetesContainerServiceImpl.java#L448)):*
<details><summary><i>Previously, abhinavsingh1196 (Abhinav Singh) wrote…</i></summary><blockquote>

ReplicaSet
</blockquote></details>

done
___
*[`960-api-services/src/main/java/io/harness/k8s/KubernetesContainerServiceImpl.java` line 480 at r1](https://reviewable.io/reviews/harness/harness-core/55115#-NissCj1DWJ0IxFdITfv:-NjgdKNZ4OoOE8i-Sy_N:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/69455926f9dfac4c094edbc27f95d509f36c8d84/960-api-services/src/main/java/io/harness/k8s/KubernetesContainerServiceImpl.java#L480)):*
<details><summary><i>Previously, abhinavsingh1196 (Abhinav Singh) wrote…</i></summary><blockquote>

DaemonSet
</blockquote></details>

done
___
*[`960-api-services/src/main/java/io/harness/k8s/KubernetesContainerServiceImpl.java` line 1420 at r1](https://reviewable.io/reviews/harness/harness-core/55115#-Nist1ZPBtojL0CW6mfL:-NjgdM3D9YaYlM9RoDQN:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/69455926f9dfac4c094edbc27f95d509f36c8d84/960-api-services/src/main/java/io/harness/k8s/KubernetesContainerServiceImpl.java#L1420)):*
<details><summary><i>Previously, abhinavsingh1196 (Abhinav Singh) wrote…</i></summary><blockquote>

nit: should be Service/%s
</blockquote></details>

done
___
*[`960-api-services/src/test/java/io/harness/k8s/KubernetesContainerServiceImplTest.java` line 1948 at r1](https://reviewable.io/reviews/harness/harness-core/55115#-NistNw04gKfZldu-FaM:-NjgdNhg4UbRP8ADs__t:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/69455926f9dfac4c094edbc27f95d509f36c8d84/960-api-services/src/test/java/io/harness/k8s/KubernetesContainerServiceImplTest.java#L1948)):*
<details><summary><i>Previously, abhinavsingh1196 (Abhinav Singh) wrote…</i></summary><blockquote>

should we also add test for 
- when all API calls fail and it returns null
- for customObjects API
?
</blockquote></details>

done

**Comment by GoelPratyush99** [comment]
trigger uts

**Pull request closed without merging**